tests/testthat/test-fixed_regex_linter.R

```diff
@@ -5,336 +5,194 @@
-    fixed_regex_linter()
-  )
-  expect_lint(
-    "str_split(x, '[\\U{F7D5}]')",
-    rex::rex("For static regular expression patterns, set `fixed = TRUE`."),
-    fixed_regex_linter()
-  )
-  expect_lint(
-    "str_split(x, '[\\U{1D4D7}]')",
-    rex::rex("For static regular expression patterns, set `fixed = TRUE`."),
-    fixed_regex_linter()
-  )
+  linter <- fixed_regex_linter()
+  msg <- rex::rex("For static regular expression patterns, set `fixed = TRUE`.")
+
+  expect_lint("gsub('[\\\\n]', '', x)", msg, linter)
+  expect_lint("gsub('[\\\\\\\"]', '', x)", msg, linter)
+  expect_lint("str_split(x, '[\\\\1]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\12]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\123]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\xa]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\xA7]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\uF]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\u01]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\u012]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\u0123]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\U8]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\U1d4d7]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\u{1}]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\U{F7D5}]')", msg, linter)
+  expect_lint("str_split(x, '[\\\\U{1D4D7}]')", msg, linter)
 })
 
 test_that("bracketed unicode escapes are caught", {
-  expect_lint(
-    'gsub("\\u{A0}", " ", out, useBytes = TRUE)',
-    rex::rex("For static regular expression patterns, set `fixed = TRUE`."),
-    fixed_regex_linter()
-  )
-  expect_lint(
-    'gsub("abc\\U{A0DEF}ghi", " ", out, useBytes = TRUE)',
-    rex::rex("For static regular expression patterns, set `fixed = TRUE`."),
-    fixed_regex_linter()
-  )
-  expect_lint(
-    'gsub("\\u{A0}\\U{0001d4d7}", " ", out, useBytes = TRUE)',
-    rex::rex("For static regular expression patterns, set `fixed = TRUE`."),
-    fixed_regex_linter()
-  )
+  linter <- fixed_regex_linter()
+  msg <- rex::rex("For static regular expression patterns, set `fixed = TRUE`.")
+
+  expect_lint('gsub("\\\\u{A0}", " ", out, useBytes = TRUE)', msg, linter)
+  expect_lint('gsub("abc\\\\U{A0DEF}ghi", " ", out, useBytes = TRUE)', msg, linter)
+  expect_lint('gsub("\\\\u{A0}\\\\U{0001d4d7}", " ", out, useBytes = TRUE)', msg, linter)
 })
 
 test_that("escaped characters are handled correctly", {
-  expect_lint("gsub('\\n+', '', sql)", NULL, fixed_regex_linter())
-  expect_lint('gsub("\\n{2,}", "\n", D)', NULL, fixed_regex_linter())
-  expect_lint('gsub("[\\r\\n]", "", x)', NULL, fixed_regex_linter())
-  expect_lint('gsub("\\n $", "", y)', NULL, fixed_regex_linter())
-  expect_lint('gsub("```\\n*```r*\\n*", "", x)', NULL, fixed_regex_linter())
-  expect_lint('strsplit(x, "(;|\n)")', NULL, fixed_regex_linter())
-  expect_lint('str_count(a$x$html, "\\\\<li\\\\>")', NULL, fixed_regex_linter())
```

Comment:
rather than remove this, we should add it to the true positive list

Reply:
NVM, let's leave this untested, per comment below: https://github.com/r-lib/lintr/pull/1032#issuecomment-1112794556